src/main/java/app/daos/impl/ReviewDAOImpl.java

```diff
@@ -0,0 +1,106 @@
+package app.daos.impl;
+
+import app.daos.AbstractDAO;
+import app.entities.Game;
+import app.entities.Review;
+import app.entities.User;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.persistence.TypedQuery;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ReviewDAOImpl extends AbstractDAO<Review, Long> {
+
+    private static ReviewDAOImpl instance;
+
+    private ReviewDAOImpl(EntityManagerFactory emf) {
+        super(emf, Review.class);
+    }
+
+    public static ReviewDAOImpl getInstance(EntityManagerFactory emf) {
+        if (instance == null) {
+            instance = new ReviewDAOImpl(emf);
+        }
+
+        return instance;
+    }
+
+    @Override
+    public Review create(Review review) {
+        try (EntityManager em = emf.createEntityManager()) {
+            User foundUser = em.find(User.class, review.getUsername());
+
+            if (foundUser == null) {
+                throw new EntityNotFoundException(String.format("User with username %s could not be found", review.getUsername()));
+            }
+
+            Game foundGame = em.find(Game.class, review.getGameId());
+
+            if (foundGame == null) {
+                throw new EntityNotFoundException(String.format("Game with id %s could not be found", review.getGameId()));
+            }
+
+            em.getTransaction().begin();
+            em.persist(review);
+
+            TypedQuery<Double> averageQuery = em.createNamedQuery("Review.getAverageRating", Double.class);
+            averageQuery.setParameter("gameId", foundGame.getId());
+
+            foundGame.setRating(averageQuery.getSingleResult());
+
+            TypedQuery<Long> countQuery = em.createNamedQuery("Review.getRatingCount", Long.class);
+            countQuery.setParameter("gameId", foundGame.getId());
+
+            foundGame.setRatingCount(countQuery.getSingleResult());
+
+            em.getTransaction().commit();
+            return review;
```

Comment:
Personal preference - would be nice to have a newline before return like we do in other methods?